auth_mellon_config.c

```diff
@@ -1139,6 +1143,52 @@ static const char *am_set_redirect_domains(cmd_parms *cmd,
     return NULL;
 }
 
+/* This function handles the MellonSignatureMethod configuration directive.
+ * This directive can be set to one of:
+ *
+ * Parameters:
+ *  cmd_parms *cmd       The command structure for this configuration
+ *                       directive.
+ *  void *struct_ptr     Pointer to the current directory configuration.
+ *  const char *arg      The string argument following this configuration
+ *                       directive in the configuraion file.
+ *
+ * Returns:
+ *  NULL on success or an error string if the argument is wrong.
+ */
+static const char *am_set_signature_method_slot(cmd_parms *cmd,
+                                                void *struct_ptr,
+                                                const char *arg)
+{
+    am_dir_cfg_rec *d = (am_dir_cfg_rec *)struct_ptr;
+
+    if (!strcasecmp(arg, "rsa-sha1")) {
+        d->signature_method = LASSO_SIGNATURE_METHOD_RSA_SHA1;
+    } else if (!strcasecmp(arg, "dsa-sha1")) {
+        d->signature_method = LASSO_SIGNATURE_METHOD_DSA_SHA1;
+    } else if (!strcasecmp(arg, "hmac-sha1")) {
+        d->signature_method = LASSO_SIGNATURE_METHOD_HMAC_SHA1;
```

Comment:
Have you tested the HMAC and DSA algorithms? HMAC is based on a shared secret, so I am curious about whether it actually works with the current configuration option for setting the SP private key. Likewise I am curious about whether DSA works.

I'm actually tempted to say that unless someone specifically needs DSA or HMAC support, there is no reason to include them.

Reply:
No I did not test the other variants, I just added what Lasso (>= 2.5.0) defined and tested with both RSA_SHA1 and RSA_SHA256. I'm fine with not including the DSA and HMAC variants at this juncture. The primary driving need was to support RSA_SHA256 (mostly because ADFS wants that). The patch now only supports the RSA_SHA* variants.